pathmux/viztree_test.go

```diff
@@ -0,0 +1,117 @@
+	t.Logf("Adding path %s", path)
+	err := tree.Add(path, true)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestVizTree(t *testing.T) {
+	tree := &Tree{path: "/"}
+
+	addPathToTree(t, tree, "/")
+	addPathToTree(t, tree, "/i")
+	addPathToTree(t, tree, "/i/:aaa")
+	addPathToTree(t, tree, "/images/abc.jpg")
+	addPathToTree(t, tree, "/images/abc.jpg/:size")
+	addPathToTree(t, tree, "/images/:imgname")
+	addPathToTree(t, tree, "/images")
+	addPathToTree(t, tree, "/images/*path")
+	addPathToTree(t, tree, "/ima")
+	addPathToTree(t, tree, "/ima/:par")
+	addPathToTree(t, tree, "/images1")
+	addPathToTree(t, tree, "/images1/*path1")
+	addPathToTree(t, tree, "/images2")
+	addPathToTree(t, tree, "/apples")
+	addPathToTree(t, tree, "/app/les")
+	addPathToTree(t, tree, "/apples1")
+	addPathToTree(t, tree, "/appeasement")
+	addPathToTree(t, tree, "/appealing")
+	addPathToTree(t, tree, "/date/:year/:month")
+	addPathToTree(t, tree, "/date/:year/month")
+	addPathToTree(t, tree, "/date/:year/:month/abc")
+	addPathToTree(t, tree, "/date/:year/:month/:post")
+	addPathToTree(t, tree, "/date/:year/:month/*post")
+	addPathToTree(t, tree, "/:page")
+	addPathToTree(t, tree, "/:page/:index")
+	addPathToTree(t, tree, "/post/:post/page/:page")
+	addPathToTree(t, tree, "/plaster")
+	addPathToTree(t, tree, "/users/:pk/:related")
+	addPathToTree(t, tree, "/users/:id/updatePassword")
+	addPathToTree(t, tree, "/:something/abc")
+	addPathToTree(t, tree, "/:something/def")
+
+	vizTree := (*VizTree)(NewVizTree(tree))
+	if !vizTree.canMatch {
+		t.Error("/ should match")
+		return
+	}
+	testChildren(t, vizTree, []string{"i", "ima", "images", "images1", "images2", "apples", "apples1", "appeasement", "appealing", "app", "date", "plaster", ":page", "post", "users", ":something"})
+	testIfChildMatch(t, vizTree, "i")
+	testChildren(t, vizTree.Child("i"), []string{":aaa"})
+	testIfChildMatch(t, vizTree, "images")
+	testChildren(t, vizTree.Child("images"), []string{"abc.jpg", "*path", ":imgname"})
+	testChildren(t, vizTree.Child("images").Child("abc.jpg"), []string{":size"})
+	testIfChildMatch(t, vizTree, "images1")
+	testChildren(t, vizTree.Child("images1"), []string{"*path1"})
+	testIfChildMatch(t, vizTree, "app")
+	testChildren(t, vizTree.Child("app"), []string{"les"})
+	testChildren(t, vizTree.Child(":something"), []string{"abc", "def"})
+	testChildren(t, vizTree.Child("users"), []string{":pk", ":id"})
+	testChildren(t, vizTree.Child("users").Child(":pk"), []string{":related"})
+	testChildren(t, vizTree.Child("users").Child(":id"), []string{"updatePassword"})
+	testChildren(t, vizTree.Child("post").Child(":post").Child("page"), []string{":page"})
+	testChildren(t, vizTree.Child("date"), []string{":year"})
+	testChildren(t, vizTree.Child("date").Child(":year"), []string{":month", "month"})
+	testChildren(t, vizTree.Child("date").Child(":year").Child(":month"), []string{":post", "abc", "*post"})
+}
+
+func testChildren(t *testing.T, tree *VizTree, expectedChildren []string) {
```

Comment:
expectedChildren is a great use case for varargs, makes testChildren() calls simpler